providers/zohocrm/metadata.go

```diff
@@ -0,0 +1,113 @@
+package zohocrm
+
+import (
+	"context"
+	"sync"
+
+	"github.com/amp-labs/connectors/common"
+	"github.com/amp-labs/connectors/common/naming"
+)
+
+// restMetadataEndpoint is the resource for retrieving metadata details.
+// doc: https://www.zoho.com/crm/developer/docs/api/v6/field-meta.html
+const restMetadataEndpoint = "settings/fields"
+
+// apiKeyField is the key holding the metadata field name.
+const apiKeyField = "api_name"
+
+type metadataFields struct {
+	Fields []map[string]any `json:"fields"`
+}
+
+func (c *Connector) ListObjectMetadata(ctx context.Context,
+	objectNames []string,
+) (*common.ListObjectMetadataResult, error) {
+	var (
+		wg sync.WaitGroup //nolint: varnamelen
+		mu sync.Mutex     //nolint: varnamelen
+	)
+
+	wg.Add(len(objectNames))
```

Comment:
This can be move to after `if len(objectNames) == 0` check since `wg` is not referred before the loop.